tests/test_automation/nightly_test_scripts/nightly_ornl.sh

```diff
@@ -393,7 +393,7 @@ if [[ $sys == *"legacycu2hip"* ]]; then
     export ROCM_PATH=/opt/rocm
     export PATH=${PATH}:${ROCM_PATH}/bin:${ROCM_PATH}/opencl/bin
     QMCPACK_TEST_SUBMIT_NAME=${QMCPACK_TEST_SUBMIT_NAME}-Legacy-CUDA2HIP
-    QMC_OPTIONS="${QMC_OPTIONS};-DENABLE_CUDA=ON;-DQMC_CUDA2HIP=ON;-DHIP_ARCH=${amdgpuarch}"
+    QMC_OPTIONS="${QMC_OPTIONS};-DENABLE_CUDA=ON;-DQMC_CUDA2HIP=ON;-DCMAKE_HIP_ARCHITECTURES=${amdgpuarch}"
```

Comment:
I'd like to see CMake get the GPU architecture then we check that the CMakeCache.txt matches what we got from the output of rocm-info.

https://github.com/Kitware/CMake/blob/f72c405d4ee4ad4775a10a944774b91f49726dad/Modules/CMakeDetermineHIPCompiler.cmake#L151

Reply:
That is kind of smart things we may try. Should be done for both CUDA and HIP.
There is long list of things to verify.
1. What happens if rocm-info is not available. This is very rare.
2. What if the login has no GPU.
3. What if the node has two GPUs but with different architectures. For example gfx1031 and gfx906 on the same machine.